packages/hawtio/src/plugins/jmx/JmxContent.tsx

```diff
@@ -39,10 +39,23 @@ export const JmxContent: React.FunctionComponent = () => {
     )
   }
 
-  const mBeanApplicable = (node: MBeanNode) => node.objectName
+  const mBeanApplicable = (node: MBeanNode) => Boolean(node.objectName)
+  const mBeanCollectionApplicable = (node: MBeanNode) => Boolean(node.children?.every(child => child.objectName))
+  const DEFAULT = (node: MBeanNode) => true
+
+  const tableSelector: (node: MBeanNode) => React.FunctionComponent = (node: MBeanNode) => {
+    const tablePriorityList: { condition: (node: MBeanNode) => boolean; element: React.FunctionComponent }[] = [
+      { condition: mBeanApplicable, element: Attributes },
+      { condition: mBeanCollectionApplicable, element: AttributeTable },
+      { condition: DEFAULT, element: JmxContentMBeans },
+    ]
+
+    //Find first
+    return tablePriorityList.filter(entry => entry.condition(node))[0].element
```

Comment:
You are doing the same thing with `array.filter`. you're not just covering the case when the return array is empty from the type point of view. instead of the current impl, the following should be more type safe:
```
return tablePriorityList.find(entry => entry.condition(node)) ?? JmxContentMBeans
```
plus, you'll no longer need `DEFAULT` const.

Reply:
I changed it to `return tablePriorityList.find(entry => entry.condition(node))?.element ?? JMXContentMBeans` as I still have to access the element to return it

I'd also still need the constant for this line for better readability https://github.com/hawtio/hawtio-next/pull/215/files#diff-64c355af4f9aa8d4446600bdc8f8295261dca01a008b626303d62019b5567c98R57 although I renamed it to ALWAYS